guillotina/json/deserialize_content.py

```diff
@@ -72,21 +72,22 @@ def __init__(self, context, request):
 
         factory = get_cached_factory(self.context.type_name)
         main_schema = factory.schema
-        await self.set_schema(
+        changed = await self.set_schema(
```

Comment:
@vangheem IMO, `self.set_schema` method should not return anything(Ideally). Does it make sense to have this flag set in the `main_schema`  object and we can just check ?
```
if main_schema.schema_changed: self.context.register()
```

Reply:
The schema isn't changing though--the value on an instance object has changed.

`main_schema` here references a global reference to a schema interface(which is a python class). So `main_schema.schema_changed` wouldn't actually quite be appropriate.

I admit that having `set_schema` return a boolean feels awkward here though.

Also, naming... `set_schema` should probably be renamed to `set_values_from_schema`

Reply:
yeah makes sense

Reply:
updated to not change return type